include/coap2/coap.h.in

```diff
@@ -40,6 +40,7 @@ extern "C" {
 #include "coap_dtls.h"
 #include "coap_event.h"
 #include "coap_io.h"
+#include "coap_tcp.h"
```

Comment:
Now the internal functions `coap_socket_connect_tcp1()` etc. are exposed via the public interface. This is problematic (not only but especially) because these functions are not compiled when TCP is disabled, probably causing linker errors for applications. (Oops, just saw who did this change...)
Hm, maybe we can move the prototype for `coap_tcp_is_supported()` somewhere else and keep this internal.

Reply:
I will move `coap_tcp_is_supported()` into coap_io.h and make coap_tcp.h coap_tcp_internal.h to get around this.